src/particleContainer/ParticleContainer.h

```diff
@@ -113,8 +113,9 @@ class ParticleContainer: public MemoryProfilable {
 	//! @param inBoxCheckedAlready - if true, spare check whether molecule is in bounding box
 	//! @param checkWhetherDuplicate - if true, check whether molecule already exists and don't insert it.
 	//! @param rebuildCaches specifies, whether the caches should be rebuild
+	//! @param updateGlbNumPrtl specifies, whether the global number of particles should be updated
 	//! @return true if successful, false if particle outside domain
-	virtual bool addParticle(Molecule& particle, bool inBoxCheckedAlready = false, bool checkWhetherDuplicate = false, const bool& rebuildCaches=false) = 0;
+	virtual bool addParticle(Molecule& particle, bool inBoxCheckedAlready = false, bool checkWhetherDuplicate = false, const bool& rebuildCaches=false, const bool updateGlbNumPrtl = false) = 0;
```

Comment:
is it really a good idea to have default values on these switches? Seems error-prone. As far as I can tell you set these values in all calls anyways?

Edit: Actually it is a really bad idea. We should not have any default arguments on virtual methods. See [Reasons and explanations](https://stackoverflow.com/questions/3533589/can-virtual-functions-have-default-parameters/3533692) and the [Google Guideline](https://google.github.io/styleguide/cppguide.html#Default_Arguments).

Reply:
Actually, I just followed the other default arguments.

Reply:
To be honest yes leaving them out is the correct way to do this, however I would say this should be done in the scope of a different PR (see my recent comment about software sustainability...)